src/mon/Monitor.cc

```diff
@@ -2903,8 +2904,20 @@ void Monitor::handle_command(MonOpRequestRef op)
 
   if (mon_cmd->is_mgr() &&
       osdmon()->osdmap.test_flag(CEPH_OSDMAP_REQUIRE_LUMINOUS)) {
-    dout(10) << __func__ << " proxying mgr command" << dendl;
-    C_MgrProxyCommand *fin = new C_MgrProxyCommand(this, op, m);
+    const auto& hdr = m->get_header();
+    uint64_t size = hdr.front_len + hdr.middle_len + hdr.data_len;
+    uint64_t max =
+      g_conf->mon_client_bytes * g_conf->mon_mgr_proxy_client_bytes_ratio;
+    if (mgr_proxy_bytes + size > max) {
+      dout(10) << __func__ << " current mgr proxy bytes " << mgr_proxy_bytes
+	       << " + " << size << " > max " << max << dendl;
+      reply_command(op, -EAGAIN, "hit limit on proxied mgr commands", rdata, 0);
```

Comment:
We may want to also test against the mgr client throttle settings, in case of dramatic variance.

Also I'm a bit concerned about the different client behavior they can see for no apparent reason. But we probably can't do much about that. :/